CONTRIBUTING.md

```diff
@@ -134,26 +141,52 @@ For additional information on using the `dep` tool, see
 
 ## Patch format
 
+### General format
+
 Beside the `Signed-off-by` footer, we expect each patch to comply with the
 following format:
 
 ```
-Subsystem: One line change summary
+subsystem: One line change summary
+
+More detailed explanation of your changes (why and how)
+that can span as many lines as required.
+
+A "Fixes #XXX" comment listing the github issue this change resolves.
+This is required for the main patch in a sequence - see the examples below.
 
-More detailed explanation of your changes: Why and how.
+Signed-off-by: <contributor@foo.com>
+```
 
-See:
-    http://chris.beams.io/posts/git-commit/
+The body of the message should not just be a continuation of the
+subject line, and is not used to extend the subject line beyond its length
+limit. The subject should be a complete sentence and the body should be a complete stand alone paragraph.
 
-For some more good advice, and the Linux Kernel document:
-    https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/SubmittingPatches
+### Subsystem
 
-Fixes: #nnn
+The "subsystem" describes the area of the code that the change applies to. It does not have to match a particular directory name in the source tree - it is just a "hint" to the reader. The subsystem is generally a single word. Although the subsystem must be specified, it is not validated: it is up to the author to decide what is a relevant subsystem for each patch.
 
-Signed-off-by: <contributor@foo.com>
+Examples:
+
+| Subsystem | Description |
+|--|--|
+| `build` | `Makefile` or configuration script change |
+| `cli` | Change affecting command line options or commands |
+| `docs` | Documentation change |
+| `logging` | Logging change |
+| `vendor` | [Re-vendoring](#re-vendor-prs) change |
+
+To see the subsystem values chosen for existing commits:
+
+```
+$ git log --no-merges --pretty="%s" | cut -d: -f1 | sort -u
 ```
 
-For example:
+### Examples
+
+#### Main patch
+
+Below is an example of a full patch description for the main change, showing the required "`Fixes #XXX`" comment which references the github issue this patch resolves:
```

Comment:
The following is an example of a full patch description for the main change that shows the required "`Fixes #XXX`" comment, which references the GitHub issue this patch resolves: